mipengine/node/monetdb_interface/common_actions.py

```diff
@@ -205,38 +205,31 @@ def get_table_data(table_name: str) -> List[ColumnData]:
     return columns_data
 
 
-def get_initial_data_schemas() -> List[str]:
+def get_initial_data_models() -> List[str]:
     """
-    Retrieves all the different schemas that the initial datasets have.
+    Retrieves all the different data_models that the initial datasets have.
 
     Returns
     ------
     List[str]
-        The dataset schemas in the database.
+        The dataset data_models in the database.
     """
 
-    schema_table_names = MonetDB().execute_and_fetchall(
-        f'SELECT code FROM "mipdb_metadata"."data_models"'
+    data_models = MonetDB().execute_and_fetchall(
+        f'SELECT code, version FROM "mipdb_metadata"."data_models"'
     )
 
-    # Flatten the list
-    schema_table_names = [
-        schema_table_name
-        for schema_table in schema_table_names
-        for schema_table_name in schema_table
-    ]
-
-    return schema_table_names
+    return data_models
```

Comment:
`data_models` here is a list of tuples with code/version.

In the system we should have only the format of `code:version`.

A "unifier" and a "splitter" should be created here, to be used from `get_initial_data_models` and `get_data_model_datasets`.